zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/tenant/DbTenantState.java

```diff
@@ -158,4 +160,19 @@ public Map<EntityType, List<Long>> getEntitiesByType(final long tenantKey) {
 
     return entitiesMap;
   }
+
+  @Override
+  public boolean forEachTenantUntilDone(final Function<String, Boolean> callback) {
+    final var done = new AtomicBoolean(false);
+    tenantsColumnFamily.whileTrue(
+        (k, p) -> {
+          done.set(!visitTenant(p.getTenantId(), callback));
+          return !done.get();
```

Comment:
Applied the same here: https://github.com/camunda/camunda/pull/25741/files#diff-f85f73dd956ccd6f1df51a7128962c2c68479b12c21daae2a7b68618a6a60300R199 - Because that's where I copied the test from and adjusted it.

Reply:
Thanks for feedback. Honestly, I was fighting with the negations. I applied your feedback (with a bit of Javadoc), and I believe it should now be easier to understand.